### DISCUSSION TITLE
Pull requests should list other pull requests that touch the same files or are likely to conflict

### DISCUSSION THREAD

**Discussion by IceCreamYou** [community/community#40482] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

When there are multiple teams of engineers working on the same codebase, or when there are multiple open source contributors that aren't planning their work together, sometimes multiple people end up working on changes that would cause git conflicts or that would interfere with each other when run. To help people realize when this will happen (in advance of when one of those PRs merges and now the other ones have to fix git conflicts), it would be helpful if PRs could display a list of other open PRs that touch any of the same files. It would also be helpful if PRs could display a list of PRs that touch any of the same files that have closed since the currently viewed PR was opened.

I realize that there are some edge cases:
- There may be some files or repos where this behavior is not desired - for example if all changes are to one big file then it is probably not that useful to say that all other PRs are also touching that file. Perhaps files to omit from this feature could be configured via a file in the `.github` folder. Maybe there should also be heuristics about files to exclude by default, such as package-lock.json, anything in a `/dist` folder, etc.
- There may be situations where too many PRs touch the same files to list them all. For example a PR that reformats the code to use a different style would likely cause conflicts with every other open PR, so there would need to be some way to elide very long lists.
- This feature would likely stop being useful on PRs that are old and inactive. Perhaps this feature could be hidden if there are already merge conflicts.

**Comment by ossanna16** [comment]
Hi there @IceCreamYou and welcome to our community! Thank you for the feedback!
